src/vtcomposite.cpp

```diff
@@ -583,61 +583,89 @@ struct InternationalizeWorker : Napi::AsyncWorker
 {
     using Base = Napi::AsyncWorker;
 
-    using property_list = std::vector<std::pair<std::string, vtzero::data_view>>; // maybe this exists in vtzero
-
     InternationalizeWorker(std::unique_ptr<InternationalizeBatonType>&& baton_data, Napi::Function& cb)
         : Base(cb),
           baton_data_{std::move(baton_data)},
           output_buffer_{std::make_unique<std::string>()} {}
 
-    std::vector<std::string> processWorldview(const std::string& pval) const
+    // for a given feature, determine how many clones of the feature are
+    // required given a worldview.
+    // - If the feature has _mbx_worldview: all, return {"all"}
+    // - If there is no requested worldview "null", determine which legacy worldviews should be created
+    std::vector<std::string> worldviews_for_feature(std::string const& pval) const
     {
-        static const std::vector<std::string> legacy_worldviews{"CN", "IN", "JP", "US"};
-
         if (pval == "all")
         {
             return {"all"};
         }
 
+        // convert pval into vector of strings US,CN => {"US", "CN"}
+        // and sort ascending
+        std::vector<std::string> worldview_values;
+        std::stringstream s_stream(pval); // create string stream from the string
+        while (s_stream.good())
+        {
+            std::string substr;
+            std::getline(s_stream, substr, ','); // get first string delimited by comma
+            worldview_values.push_back(substr);
+        }
+        std::sort(worldview_values.begin(), worldview_values.end());
+
+        // result will be a vector of matching worldview values
+        std::vector<std::string> result;
+
+        // worldview: null
         if (baton_data_->worldview.empty())
         {
-            std::vector<std::string> legacy;
-            for (auto const& wv : legacy_worldviews)
-            {
-                if (pval.find(wv) != std::string::npos)
-                {
-                    legacy.push_back(wv);
-                }
-            }
-            return legacy;
+            static const std::vector<std::string> legacy_worldviews{"CN", "IN", "JP", "US"}; // MUST BE IN ALPHABETICAL ORDER
+            // {"US", "CN"} AND {"US", "IN"} => {"US"}
+            std::set_intersection(
+                worldview_values.begin(),
+                worldview_values.end(),
+                legacy_worldviews.begin(),
+                legacy_worldviews.end(),
+                std::back_inserter(result));
         }
+        // worldview: XX
         else
         {
-            if (pval.find(baton_data_->worldview) != std::string::npos)
+            if (std::find(std::begin(worldview_values), std::end(worldview_values), baton_data_->worldview) != std::end(worldview_values))
```

Comment:
maybe worldview_values.begin() and .end() would be more idiomatic? But just a style choice

Reply:
I agree I like `begin()` and `end()` more. This was copied from another usage of `std::end`. I've updated the entirety of vtcomposite.cpp to use begin/end now!